Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs

```diff
@@ -1019,17 +1023,30 @@ private static void QueryScene(IMixedRealityPointer pointer, IMixedRealityRaycas
                         }
                         break;
                     case SceneQueryType.SphereOverlap:
-                        Collider[] colliders = UnityEngine.Physics.OverlapSphere(pointer.Rays[i].Origin, pointer.SphereCastRadius, ~UnityEngine.Physics.IgnoreRaycastLayer);
 
-                        if (colliders.Length > 0)
+                        int numColliders;
+
+                        if (colliderBuffer != null)
+                        {
+                            numColliders = UnityEngine.Physics.OverlapSphereNonAlloc(pointer.Rays[i].Origin, pointer.SphereCastRadius, colliderBuffer, ~UnityEngine.Physics.IgnoreRaycastLayer);
+                        }
+                        else
+                        {
+                            colliderBuffer = UnityEngine.Physics.OverlapSphere(pointer.Rays[i].Origin, pointer.SphereCastRadius, ~UnityEngine.Physics.IgnoreRaycastLayer);
+                            numColliders = colliderBuffer.Length;
+                        }                        
+
+                        if (colliderBuffer.Length > 0)
                         {
                             Vector3 testPoint = pointer.Rays[i].Origin;
                             GameObject closest = null;
                             float closestDistance = Mathf.Infinity;
                             Vector3 objectHitPoint = testPoint;
 
-                            foreach (Collider collider in colliders)
+                            for (int c = 0; c < numColliders; c++)
```

Comment:
Why not foreach?

Reply:
Removed the changes here since they aren'trelevant to the bugfix. Let's re-open the discussion in separate issue if needed.